language-support/codegen-common/src/main/scala/com/digitalasset/daml/lf/codegen/conf/CodegenConfigReader.scala

```diff
@@ -158,17 +158,20 @@ object CodegenConfigReader {
   implicit val decodePackageReference: KeyDecoder[PackageReference] =
     new KeyDecoder[PackageReference] {
       // TODO (MK) https://github.com/digital-asset/daml/issues/9934
-      // For now we only allow name-vesion pairs to match the compiler. Once the compiler
+      // For now we only allow name-version pairs to match the compiler. Once the compiler
       // accepts package ids we can allow for those here as well.
       final def apply(key: String): Option[PackageReference] = nameVersion(key)
 
-      private def nameVersion(key: String): Option[PackageReference] = key.split("-") match {
-        case Array(name, version) =>
+      private def nameVersion(key: String): Option[PackageReference] = {
+        val dashIndex = key.lastIndexWhere(_ == '-')
+        if (dashIndex < 0) {
+          None
+        } else {
           for {
-            name <- PackageName.fromString(name).toOption
-            version <- PackageVersion.fromString(version).toOption
+            name <- PackageName.fromString(key.slice(0, dashIndex)).toOption
+            version <- PackageVersion.fromString(key.slice(dashIndex + 1, key.length)).toOption
           } yield PackageReference.NameVersion(name, version)
-        case _ => None
+        }
```

Comment:
I don't understand the need for a private method that is just invoked, you can probably do something like the following:
```
    key => {
      key.splitAt(key.lastIndexOf('-')) match {
        case (rawPackageName, rawPackageVersion)
            if rawPackageName.nonEmpty && rawPackageVersion.nonEmpty =>
          for {
            name <- PackageName.fromString(rawPackageName).toOption
            version <- PackageVersion.fromString(rawPackageVersion).toOption
          } yield PackageReference.NameVersion(name, version)
        case _ =>
          None
      }
```

Reply:
Inlined it, I don’t think your suggestion works. You are not dropping the dash in between. I also prefer handling the fact where there is no dash explicitly instead of relying on PackageName.fromString to blow up on an empty name (which you get for splitting at -1).

Reply:
> relying on PackageName.fromString to blow up on an empty name (which you get for splitting at -1)

I'm not sure I follow this, the guard clearly prevents you from going into that branch if either is an empty string.

Reply:
Oh sorry right I missed that guard. You still have the issue that you’re not dropping the dash.

Reply:
switched to something closer to what you suggested, I find that harder to follow but your choice.

Reply:
Yes, I agree that that's hard to follow. I didn't want you to go in that direction but I don't want you to bug any further, just merge as is, I'll change it afterwards.

Reply:
apologies for being a bit slow in following what you were suggesting, I’ll merge as you suggested and let you clean it up. Sorry for the extra work caused by that.

Reply:
By the way, your code was silently relying on `PackageName.fromString` to blow up on an empty name or version (i.e. a dash at the beginning or end of the string).

Reply:
good point

Reply:
Took care of it in https://github.com/digital-asset/daml/pull/12982